### PR TITLE
Refactor e2e page objects into logical clusters

### PR DESCRIPTION
**Pull request by fschade**
## Description

The main reason for this pr is that the tests now have to support some actions on different pages, not only on the personal page as before.

as usual it's more than expected and fixes some bugs and flaky test steps.

In a nutshell it contains:
* new debounced rev composable
* cleanup the feature file step wording
* unify some test steps into one and use the table to transport information
* fix sharee sidebar which rendered multiple times in cases where the backend requests took a bit longer
* reset collaborators select loading state and collaborators list after adding sharee
* differentiate between functional bdd tests (we call them integrations from now on) and journeys (storytelling tests)
* refactor playwright test actions into logical clusters (resource, share, user, ...) instead of page groups (actions for all-files, shared-with-me, ...)
* add option to test anonymous users and keep reference of them
* add option to store public links (many, not only last) and re-use them later 

## Related Issue
- Fixes https://github.com/owncloud/web/issues/6596

## Motivation and Context
take care of our tests ;)

## How Has This Been Tested?
- tests (unit, integration, e2e)

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/23711/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23711/tracing/file-alice-2022-3-15-04-22-13.zip 


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntGroupsToRoot</strong> https://drone.owncloud.com/owncloud/web/23731/28/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingInternalGroupsToRoot-shareWithGroups_feature-L21.png
  

### webUISharingInternalGroupsToRootEdgeCases-shareWithGroupsEdgeCases_feature-L40.png
  

### webUISharingInternalGroupsToRootEdgeCases-shareWithGroupsEdgeCases_feature-L56.png
  

### webUISharingInternalGroupsToRootEdgeCases-shareWithGroupsEdgeCases_feature-L72.png
  

### webUISharingInternalGroupsToRootEdgeCases-shareWithGroupsEdgeCases_feature-L88.png
  


</p></details>
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: packages/web-app-files/src/components/SideBar/Shares/FileShares.vue [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
     <oc-loader v-if="sharesLoading" :aria-label="$gettext('Loading people list')" />
     <template v-else>
-      <invite-collaborator-form v-if="currentUserCanShare" key="new-collaborator" class="oc-my-s" />
-      <p
-        v-else
-        key="no-reshare-permissions-message"
-        data-testid="files-collaborators-no-reshare-permissions-message"
-        v-text="noResharePermsMessage"
-      />
       <div v-if="hasSharees" class="avatars-wrapper oc-flex oc-flex-middle oc-flex-between">
```
Comment (saw-jan) [inline packages/web-app-files/src/components/SideBar/Shares/FileShares.vue:12]: this change fixes some intermittent fails in e2e such as this issue https://github.com/owncloud/web/issues/6596#issuecomment-1068722941

Comment (fschade) [reply packages/web-app-files/src/components/SideBar/Shares/FileShares.vue:12]: yes, but this should not be the solutions..... it's a nasty behavior by web :( i have to think how to fix that best.

Comment (fschade) [reply packages/web-app-files/src/components/SideBar/Shares/FileShares.vue]: @saw-jan fixed it, lets see if the ci likes it ;)

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/23763/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23763/tracing/folder-alice-2022-3-16-02-10-11.zip 


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic2</strong> https://drone.owncloud.com/owncloud/web/23766/68/1
The following scenarios passed on retry:
- webUISharingPublicDifferentRoles/shareByPublicLinkDifferentRoles.feature:159

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingSharingInd</strong> https://drone.owncloud.com/owncloud/web/23766/32/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingInternalGroupsSharingIndicator-shareWithGroups_feature-L136.png
  

### webUISharingInternalGroupsToRootSharingIndicator-shareWithGroups_feature-L124.png
  

### webUISharingInternalUsersSharingIndicator-shareWithUsers_feature-L17.png
  

### webUISharingInternalUsersToRootSharingIndicator-shareWithUsers_feature-L118.png
  


</p></details>
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles3</strong> https://drone.owncloud.com/owncloud/web/23772/57/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp1</strong> https://drone.owncloud.com/owncloud/web/23773/71/1
The following scenarios passed on retry:
- webUIPrivateLinks/accessingPrivateLinks.feature:9

**Review by pascalwengerter — APPROVED** [review]
Badaboom! 💥 (We'll have to add bugfix changelog items for the Sidebar-PeopleShare improvements later:)

**Pull request merged**